packages/editor-ui/src/views/WorkflowHistory.vue

```diff
@@ -1,19 +1,147 @@
 <script setup lang="ts">
+import { saveAs } from 'file-saver';
+import { onBeforeMount, ref, watchEffect } from 'vue';
+import { useRoute, useRouter } from 'vue-router';
+import { VIEWS } from '@/constants';
 import { useI18n } from '@/composables';
+import type { TupleToUnion } from '@/utils/typeHelpers';
+import type { WorkflowHistoryActionTypes, WorkflowHistory } from '@/types/workflowHistory';
+import WorkflowHistoryList from '@/components/WorkflowHistory/WorkflowHistoryList.vue';
+import WorkflowHistoryContent from '@/components/WorkflowHistory/WorkflowHistoryContent.vue';
+import { useWorkflowHistoryStore } from '@/stores/workflowHistory.store';
 
+type WorkflowHistoryActionRecord = {
+	[K in Uppercase<TupleToUnion<WorkflowHistoryActionTypes>>]: Lowercase<K>;
+};
+
+const workflowHistoryActionTypes: WorkflowHistoryActionTypes = [
+	'restore',
+	'clone',
+	'open',
+	'download',
+];
+const WORKFLOW_HISTORY_ACTIONS: WorkflowHistoryActionRecord = workflowHistoryActionTypes.reduce(
+	(record, key) => ({ ...record, [key.toUpperCase()]: key }),
+	{} as WorkflowHistoryActionRecord,
+);
```

Comment:
1. Please use the generic instead of asserting.
2. No need to type the result, it'll flow from `reduce`.